osquery/tables/system/darwin/signature.mm

```diff
@@ -91,92 +91,95 @@ void genSignatureForFile(const std::string& path, QueryData& results) {
+
+      // Get CDHash
+      r["cdhash"] = "";
+      CFTypeRef hashInfo = CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
+      if (hashInfo != nullptr) {
+        r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
+        // Get the SHA-1 bytes
+        std::stringstream ss;
+        auto bytes = CFDataGetBytePtr((CFDataRef)hashInfo);
+        if (bytes != nullptr) {
+          // Write bytes as hex strings
+          for (unsigned n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
+            ss << std::hex << std::setfill('0') << std::setw(2);
+            ss << (unsigned int)bytes[n];
           }
+          r["cdhash"] = ss.str();
         }
-
-        // Team Identifier
-        r["team_identifier"] = "";
-        CFTypeRef teamIdent = nullptr;
-        if (CFDictionaryGetValueIfPresent(codeInfo, kSecCodeInfoTeamIdentifier,
-                                          &teamIdent)) {
-          r["team_identifier"] = stringFromCFString((CFStringRef)teamIdent);
+        if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
+          VLOG(1) << "Error extracting code directory hash";
+          r["cdhash"] = "";
         }
+      }
+
+      // Team Identifier
+      r["team_identifier"] = "";
+      CFTypeRef teamIdent = nullptr;
+      if (CFDictionaryGetValueIfPresent(codeInfo, kSecCodeInfoTeamIdentifier,
+                                        &teamIdent)) {
+        r["team_identifier"] = stringFromCFString((CFStringRef)teamIdent);
+      }
 
-        // Get common name
-        r["authority"] = "";
-        CFArrayRef certChain = (CFArrayRef)CFDictionaryGetValue(
-            codeInfo, kSecCodeInfoCertificates);
-        if (certChain != nullptr) {
-          CFIndex count = CFArrayGetCount(certChain);
-          // Only look at the first cert, which is the developer's cert
-          if (count > 0) {
-            auto cert = SecCertificateRef(CFArrayGetValueAtIndex(certChain, 0));
-            auto der_encoded_data = SecCertificateCopyData(cert);
-            if (der_encoded_data != nullptr) {
-              auto der_bytes = CFDataGetBytePtr(der_encoded_data);
-              auto length = CFDataGetLength(der_encoded_data);
-              auto x509_cert = d2i_X509(nullptr, &der_bytes, length);
-              if (x509_cert != nullptr) {
-                std::string subject;
-                std::string issuer;
-                std::string commonName;
-                genCommonName(x509_cert, subject, commonName, issuer);
-                r["authority"] = commonName;
-                X509_free(x509_cert);
-              } else {
-                VLOG(1) << "Error decoding DER encoded certificate";
-              }
-              CFRelease(der_encoded_data);
+      // Get common name
+      r["authority"] = "";
+      CFArrayRef certChain =
+          (CFArrayRef)CFDictionaryGetValue(codeInfo, kSecCodeInfoCertificates);
+      if (certChain != nullptr) {
+        CFIndex count = CFArrayGetCount(certChain);
```

Comment:
Since `count` isn't used anywhere else, could combine the conditional `if (certChain != nullptr && CFArrayGetCount(certChain) > 0)`. Gets rid of one nested `if`.